### PR TITLE
[Relay] Move compiler_begin/end_op to local static objects

### PR DESCRIPTION
**Pull request by hlu1**
Hit compilation error:
```
[op.cc:66] Check failed: reg != nullptr Operator annotation.compiler_begin is not registered
```
It's because `compiler_begin_op` and `compiler_end_op` are non-local static objects and the relay ops must to be registered earlier, which is not guaranteed. After moving them to local static objects, the operator lookup happens at run time, after everything's initialized and registered.

### CODE REVIEW

**Comment by hlu1** [comment]
@zhiics, @tqchen, please review

File: src/relay/transforms/pass_util.h [hunk 1]
```diff
@@ -115,6 +115,20 @@ inline bool IsAtomic(const Expr& e) {
   return e.as<VarNode>() || e.as<OpNode>() || e.as<ConstructorNode>() || e.as<GlobalVarNode>();
 }
 
+/*!
+ * \brief Cache the compiler_begin annotation op to reduce registry lookup overhead
+ * \param void
+ * \return compiler_begin op
+ */
+const Op& compiler_begin_op();
```
Comment (tqchen) [inline src/relay/transforms/pass_util.h:123]: Use CamelCase

File: src/runtime/object.cc [hunk 2]
```diff
@@ -24,6 +24,7 @@
 #include <tvm/runtime/object.h>
 #include <tvm/runtime/registry.h>
 
+#include <iostream>
```
Comment (tqchen) [inline src/runtime/object.cc:27]: is this change necessary?

File: src/relay/analysis/util.cc [hunk 3]
```diff
@@ -454,5 +454,15 @@ bool IsDataDependant(const CallNode* call) {
 
   return tshape_data_dependant[op];
 }
+
+const Op& compiler_begin_op() {
```
Comment (zhiics) [inline src/relay/analysis/util.cc:458]: we probably don't want to put them under analysis namespace because they are used in the transform namespace. Can we just put them in the pass_util.h file and inline?

File: src/runtime/object.cc (hunk 2 above)
Comment (hlu1) [reply src/runtime/object.cc:27]: This is to fix another compilation error with clang:
```
stderr: tvm/tvm/src/runtime/object.cc:171:14: error: missing '#include <iostream>'; declaration of 'cerr' must be imported from module 'third_party_libgcc_std.iostream' before it is required
        std::cerr << '[' << info.index << "] " << info.name
             ^
third-party-buck/platform007/build/libgcc/include/c++/7.3.0/iostream:62:18: note: previous declaration is here
  extern ostream cerr;          /// Linked to standard error (unbuffered)
                 ^
1 error generated.
    When running <c++ preprocess_and_compile>.
    When building rule //tvm:tvm#compile-pic-object.cc.od8dbd3a8,platform007-clang.
```
Should I split it into another PR instead?

File: src/relay/transforms/pass_util.h (hunk 1 above)
Comment (hlu1) [reply src/relay/transforms/pass_util.h:123]: Yes

File: src/relay/analysis/util.cc (hunk 3 above)
Comment (hlu1) [reply src/relay/analysis/util.cc:458]: Got it

File: src/relay/analysis/util.cc [hunk 4]
```diff
@@ -338,13 +338,13 @@ bool IsNDArrayAllGreaterEqual(const runtime::NDArray& tensor, T value) {
...
-static const auto& transpose_op = Op::Get("transpose");
-static const auto& squeeze_op = Op::Get("squeeze");
-
 bool IsAllPositiveConstant(const Expr& expr) {
+  // Cache the operators that are checked recursively to reduce lookup overhead.
+  static const auto& expand_dims_op = Op::Get("expand_dims");
+  static const auto& reshape_op = Op::Get("reshape");
+  static const auto& transpose_op = Op::Get("transpose");
+  static const auto& squeeze_op = Op::Get("squeeze");
+
```
Comment (hlu1) [inline src/relay/analysis/util.cc:347]: Change these static ops from non-local to local static objects as well.

**Comment by zhiics** [comment]
Thanks @hlu1 @tqchen
